### PR TITLE
[FEAT] Kingdom Chores

### PR DESCRIPTION
**Pull request by Rofrtd**
# Description

Implements Faction Chores. All faction chores reward marks.

**Features**

1. Complete Faction Chore
2. Skip Faction Chore (once every 24 hours)
3. Automatic Refresh of Faction Chores

## How to Test

**Complete Faction Chore**

1. Load Chores
2. Make progress
3. Complete Chore

**Skip Faction Chore**

1. Load Chores
2. Skip
3. Confirm Modal

**Refresh Faction Chore**

1. Test Details in backend implementation

## Screenshots

**Complete Faction Chore**

complete_mobile
complete_desktop

**Skip Faction Chore**

skip_mobile
skip_desktop

**Refresh Faction Chore**

loading1
Loading2

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

localhost

# Checklist:

- [X] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [X] Screenshot if it includes any UI changes
- [X] I have read the contributing guidelines and agree to the T&Cs
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/ui/Box.tsx [hunk 1]
```diff
@@ -51,6 +52,14 @@ export interface BoxProps {
...
    */
   parentDivRef?: React.RefObject<HTMLElement>;
+  /**
+   * progress bar for the box, replaces the bottom left and bottom right
+   */
+  progress?: {
+    label?: string;
+    percentage: number;
+    type: ProgressType;
+  };
```
Comment (c-py) [inline src/components/ui/Box.tsx:62]: This component may be getting a bit overloaded. There's possibly a better way to do this, or just display the progress in general.

File: src/styles.css [hunk 2]
```diff
@@ -1281,3 +1281,29 @@ a {
...
+  50% {
+    content: "..";
+  }
+  75% {
+    content: "...";
+  }
+  100% {
+    content: "";
+  }
+}
```
Comment (c-py) [inline src/styles.css:1309]: I would like some feedback on this loader, perhaps a better way to do this as well.

Comment (adamhannigan) [reply src/styles.css:1309]: I faced some issues with the CSS approach to this when changing fonts.

We've got the `<Loading />` component which does this programatically using JS.

**Review by adamhannigan — APPROVED** [review]
Nice work mate!

Left comments in person. Non blocking but nice to haves.

**Pull request merged**